src/traces/scattermapbox/convert.js

```diff
@@ -369,11 +370,58 @@ function arrayifyAttribute(values, step) {
 
 function getTextFont(trace) {
     var font = trace.textfont;
-    var str = '';
-    if(font.weight === 'bold') str += ' Bold';
-    if(font.style === 'italic') str += ' Italic';
-    var textFont = font.family;
-    if(str) textFont = textFont.replace(' Regular', str);
-    textFont = textFont.split(', ');
+    var family = font.family;
+    var style = font.style;
+    var weight = font.weight;
+
+    var parts = family.split(' ');
+    var isItalic = parts[parts.length - 1] === 'Italic';
+    if(isItalic) parts.pop();
+    isItalic = isItalic || style === 'italic';
+
+    var str = parts.join(' ');
+    if(weight === 'bold' && parts.indexOf('Bold') === -1) {
+        str += ' Bold';
+    } else if(weight <= 1000) { // numeric font-weight
+        // See supportedFonts
+
+        if(parts[0] === 'Metropolis') {
+            str = 'Metropolis';
+            if(weight > 850) str += ' Black';
+            else if(weight > 750) str += ' Extra Bold';
+            else if(weight > 650) str += ' Bold';
+            else if(weight > 550) str += ' Semi Bold';
+            else if(weight > 450) str += ' Medium';
+            else if(weight > 350) str += ' Regular';
+            else if(weight > 250) str += ' Light';
+            else if(weight > 150) str += ' Extra Light';
+            else str += ' Thin';
+        } else if(parts[0] === 'Open' && parts[1] === 'Sans') {
+            str = 'Open Sans';
+            if(weight > 750) str += ' Extrabold';
+            else if(weight > 650) str += ' Bold';
+            else if(weight > 550) str += ' Semibold';
+            else if(weight > 350) str += ' Regular';
+            else str += ' Light';
+        } else { // Other families
+            str = 'Klokantech Noto Sans';
```

Comment:
Doesn't this `else` mean the font will default to `Klokantech Noto Sans` if it's not `Metropolis` or `Open Sans`?

Reply:
oh I see, maybe not, because it has already passed through the `supplyDefaults` step -- can you confirm @archmoj ?

Reply:
Good point. No that shouldn't happen because further down we test and ensure the font is supported. 
I could restrict this `else` to make it read better?

Reply:
Addressed in [bff00ac](https://github.com/plotly/plotly.js/pull/6990/commits/bff00ac8d60f0b154aaec75615621e63cb89c2f1).